astropy/coordinates/angle_utilities.py

```diff
@@ -802,3 +155,102 @@ def offset_by(lon, lat, posang, distance):
     outlat = Angle(np.arcsin(cos_b), u.radian).to(u.deg)
 
     return outlon, outlat
+
+
+def golden_spiral_grid(size):
+    """Generate a grid of points on the surface of the unit sphere using the
+    Fibonacci or Golden Spiral method.
+
+    .. seealso::
+
+        `Evenly distributing points on a sphere <https://stackoverflow.com/questions/9600801/evenly-distributing-n-points-on-a-sphere>`_
+
+    Parameters
+    ----------
+    size : int
+        The number of points to generate.
+
+    Returns
+    -------
+    rep : `~astropy.coordinates.UnitSphericalRepresentation`
+        The grid of points.
+    """
+    golden_r = (1 + 5**0.5) / 2
+
+    grid = np.arange(0, size, dtype=float) + 0.5
+    lon = 2*np.pi / golden_r * grid * u.rad
+    lat = np.arcsin(1 - 2 * grid / size) * u.rad
+
+    return UnitSphericalRepresentation(lon, lat)
+
+
+def uniform_spherical_random_surface(size=1, rng=None):
```

Comment:
I wonder if it is truly desirable to take the random number generator as an option? Other methods throughout Astropy that use an RNG do not take such an argument (e.g. `LombScargle.false_alarm_probability`). Various examples throughout the docs already suggest using `astropy.utils.NumpyRNGContext` to control the random state.

Reply:
I personally like this idiom in my own code but I don't feel too strongly here

Reply:
I don't feel strongly about this idiom either, my only concern is consistency with other modules.

Reply:
You raise a good point there @lpsinger about consistency with `NumpyRNGContext`!  I'm going to just make a command decision to remove it for now - we can always put it in in a future version.

Reply:
Ohh, I also just realized `np.random.default_rng` does *not* respect `np.random.seed`, which is what everything else assumes, so definitely important to do this for consistency.